spec/rester_spec.rb

```diff
@@ -128,4 +130,113 @@
       end
     end # without circuit_breaker_enabled
   end # ::connect
+
+  describe '::service_name' do
+    subject { Rester.service_name }
+
+    it 'should default to the Rails application name' do
+      is_expected.to eq "Dummy"
+    end
+
+    context 'with service_name set' do
+      before { Rester.service_name = "New Consumer Name" }
+      it { is_expected.to eq "New Consumer Name" }
+    end # with service_name set
+  end # ::service_name
+
+  describe '::begin_request' do
+    subject { Rester.request_info }
+    before { Rester.begin_request }
+    it { is_expected.to eq({}) }
+  end # ::begin_request
+
+  describe '::end_request' do
+    subject { Rester.request_info }
+    before { Rester.end_request }
+    it { is_expected.to eq nil }
+  end # ::end_request
+
+  describe '::request_info=' do
+    subject { Rester.request_info }
+    before { Rester.request_info = request_info }
+    let(:request_info) {  { correlation_id: id } }
+    let(:id) { SecureRandom.uuid }
+
+    context 'with single thread' do
+      context 'with new request_info' do
+        it 'should set the request info value' do
+          expect(subject[:correlation_id]).to eq id
+        end
+      end # with new request_info
+
+      context 'with nil request_info' do
+        it 'should delete the request_info for this thread' do
+          expect(subject[:correlation_id]).to eq(id)
+          subject = nil
+          expect(subject).to eq nil
+        end
```

Comment:
`subject = nil` isn't going to call `Rester.request_info=`.  It sets a local `subject` variable to `nil`, so the `expect` clause is guaranteed to be true.

I think there should be a `let(:request_info) { nil }` at the top of this context instead.